applications/asset_tracker_v2/doc/modem_module.rst

```diff
@@ -0,0 +1,163 @@
+* :c:enum:`MODEM_EVT_BATTERY_DATA_NOT_READY`
+
+Neighbor cell measurements
+==========================
+
+Neighbor cell measurements can be requested by the application by sending an ``APP_EVT_DATA_GET`` event where ``APP_DATA_NEIGHBOR_CELLS`` is part of the requested data list.
+Upon reception of that event, the modem module uses the LTE link controller to start a neighbor cell  search of type :c:enum:`LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT`.
+See :ref:`lte_lc_readme` for more details on the available search types.
+When the search completes, the module sends a :c:enum:`MODEM_EVT_NEIGHBOR_CELLS_DATA_READY` event containing the cell information received from the modem.
+If the search fails, a :c:enum:`MODEM_EVT_NEIGHBOR_CELLS_DATA_NOT_READY` event is sent.
+
+Module internals
+****************
+
+The modem module has an internal thread with a message queue for processing.
+When an event is received in the :ref:`event_manager` handler, the event is converted to a message and put into the module's queue for processing in thread context.
+This gives the module the flexibility to call functions that might take some time to complete.
+
+Module states
+=============
+
+The modem module has an internal state machine with the following states:
+
+  * ``STATE_INIT`` - The initial state of the module in which it awaits the modem to be initialized.
+    The module enters this state only if the :ref:`liblwm2m_carrier_readme` library is enabled because the modem is automatically initialized otherwise.
+  * ``STATE_DISCONNECTED`` - The module has performed all required initializations and is ready to establish an LTE connection.
+    This is the initial state for applications that do not use the :ref:`liblwm2m_carrier_readme` library.
+  * ``STATE_CONNECTING`` - The modem is currently searching for a suitable LTE network and attempting to establish a connection.
+  * ``STATE_CONNECTED`` - The device is connected to an LTE network.
+  * ``STATE_SHUTDOWN`` - The module has been shut down after receiving a request from the utility module.
+
+State transitions take place based on input from other modules through the event manager handler and the LTE link controller handler.
+
+Configuration options
+*********************
+
+You can set the following options to configure the modem module:
+
+.. _CONFIG_MODEM_MODULE:
+
+CONFIG_MODEM_MODULE - Configuration for modem module
+   This option enables the modem module.
+
+.. _CONFIG_MODEM_THREAD_STACK_SIZE:
+
+CONFIG_MODEM_THREAD_STACK_SIZE -  Configuration for thread stack size
+   This option configures the modem module thread stack size.
+
+.. _CONFIG_MODEM_SEND_ALL_SAMPLED_DATA:
+
+CONFIG_MODEM_SEND_ALL_SAMPLED_DATA - Configuration for sending all sampled data
+   By default, the modem module sends only events with sampled data that has changed since the last sampling.
+   To send unchanged data also, enable :ref:`CONFIG_MODEM_SEND_ALL_SAMPLED_DATA <CONFIG_MODEM_SEND_ALL_SAMPLED_DATA>`.
```

Comment:
Not sure whether it is reasonable to link to the Kconfig option itself.

Reply:
removed the link.